src/include/txn_inline.h

```diff
@@ -641,23 +692,30 @@ __wt_txn_pinned_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *pinned_tsp)
     WT_TXN_GLOBAL *txn_global;
     wt_timestamp_t checkpoint_ts, pinned_ts;
 
-    *pinned_tsp = WT_TS_NONE;
-
     txn_global = &S2C(session)->txn_global;
 
     /*
      * There is no need to go further if no pinned timestamp has been set yet.
      */
-    if (!txn_global->has_pinned_timestamp)
+    if (!txn_global->has_pinned_timestamp) {
```

Comment:
I found a bug in the pinned timestamp implementation ([WT-13849](https://jira.mongodb.org/browse/WT-13849)). Here I'm just copying the fix from develop.